packages/e2e-tests/specs/editor/various/multi-block-selection.test.js

```diff
@@ -491,4 +491,56 @@ describe( 'Multi-block selection', () => {
 
 		expect( await getEditedPostContent() ).toMatchSnapshot();
 	} );
+
+	it( 'should place the caret at the end of last pasted paragraph (paste to empty editor)', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'first paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'second paragraph' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await pressKeyWithModifier( 'primary', 'c' );
+		await page.keyboard.press( 'Backspace' );
+		await pressKeyWithModifier( 'primary', 'v' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+	} );
+
+	it( 'should place the caret at the end of last pasted paragraph (paste mid-block)', async () => {
+		await clickBlockAppender();
+		await page.keyboard.type( 'first paragraph' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( 'second paragraph' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await pressKeyWithModifier( 'primary', 'a' );
+		await pressKeyWithModifier( 'primary', 'c' );
+
+		await page.keyboard.press( 'ArrowRight' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+		await page.keyboard.press( 'ArrowLeft' );
+
+		await pressKeyWithModifier( 'primary', 'v' );
+		await page.keyboard.press( 'Backspace' );
+
+		expect( await getEditedPostContent() ).toMatchSnapshot();
+		// This code path triggers an unrelated React warning. Without the check below,
```

Comment:
It warns with development version of React so it won’t trigger on Travis. You can disable those warnings locally by setting `WP_DEBUG` to false. 

In short, you will have to remove this assertion to make Travis green.

Reply:
Related docs:
https://github.com/WordPress/gutenberg/blob/master/packages/env/README.md#wp-envjson

I assumed you are using wp-env.

Reply:
TIL, thank you!

Reply:
Ideally, there are no warnings on JS console, the last time it was the case around WordPress 5.0 😅